Add reference to Beats config examples in the docs (#3447)
elastic/cloud-on-k8s: docs/orchestrating-elastic-stack-applications/beat.asciidoc

EOF
----

> For more ready-to-use manifest examples, please see <<{p}-beat-configuration-examples,Configuration Examples>>.

Reviewer: ```suggestion
See <<{p}-beat-configuration-examples>> for more ready-to-use manifests.
```

To deploy Beats in clusters with the Pod Security Policy admission controller enabled, or in OpenShift clusters, you must grant additional permissions to the Service Account used by the Beat Pods. Those Service Accounts must be bound to a Role or ClusterRole that has `use` permission for the required Pod Security Policy or Security Context Constraints. Different Beats and their features might require different settings set in their PSP/SCC.

[id="{p}-beat-configuration-examples"]
== Configuration Examples

> This directory contains yaml manifests with example configurations for Beats. These manifests are self-contained and work out-of-the-box on any non-secured Kubernetes cluster. All of them contain three-node Elasticsearch cluster and single Kibana instance. All Beat configurations set up Kibana dashboards if they are available for a given Beat and all required RBAC resources.

Reviewer 2: `directory` does not make sense here unless you add a link to the actual directory

==== Metricbeat for Kubernetes monitoring

> 
> ```yaml
> kubectl apply -f https://raw.githubusercontent.com/elastic/cloud-on-k8s/1.2/config/recipes/beats/metricbeat_hosts.yaml
> ```

Reviewer: I would recommend doing something like the following where you define a `beats_url` attribute and use that instead of repeating the URL in every sample.

```suggestion
:beats_url: https://raw.githubusercontent.com/elastic/cloud-on-k8s/{eck_release_branch}/config/recipes/beats

[source,sh,subs="attributes"]
----
kubectl apply -f {beats_url}/metricbeat_hosts.yaml
----

```

Deploys Metricbeat as a DaemonSet that monitors the host resource usage (cpu, memory, network, filesystem) and Kubernetes resources (Nodes, Pods, Containers, Volumes).

==== Filebeat with autodiscover

> ```yaml

Reviewer 2: ```suggestion
```sh
```

To deploy Beats in clusters with the Pod Security Policy admission controller enabled, or in OpenShift clusters, you must grant additional permissions to the Service Account used by the Beat Pods. Those Service Accounts must be bound to a Role or ClusterRole that has `use` permission for the required Pod Security Policy or Security Context Constraints. Different Beats and their features might require different settings set in their PSP/SCC.

[id="{p}-beat-configuration-examples"]
== Configuration Examples

> Below you can find manifests that address a number of common use cases and can be your starting point in exploring Beats deployed with ECK. These manifests are self-contained and work out-of-the-box on any non-secured Kubernetes cluster. All of them contain three-node Elasticsearch cluster and single Kibana instance. All Beat configurations set up Kibana dashboards if they are available for a given Beat and all required RBAC resources.

Reviewer 3: ```suggestion
Below you can find manifests that address a number of common use cases and can be your starting point in exploring Beats deployed with ECK. These manifests are self-contained and work out-of-the-box on any non-secured Kubernetes cluster. They all contain three-node Elasticsearch cluster and single Kibana instance. All Beat configurations set up Kibana dashboards if they are available for a given Beat and all required RBAC resources.
```

> Below you can find manifests that address a number of common use cases and can be your starting point in exploring Beats deployed with ECK. These manifests are self-contained and work out-of-the-box on any non-secured Kubernetes cluster. All of them contain three-node Elasticsearch cluster and single Kibana instance. All Beat configurations set up Kibana dashboards if they are available for a given Beat and all required RBAC resources.

Reviewer 4: It's unrelated to that particular PR but I'm thinking about it now: if we consider the Elasticsearch resource of each example as not ready for production (mostly because small resource reqs and small storage size), then why do we deploy 3 instances? Wouldn't 1 instance be good enough, and more in line with how this is an non-production example?

Author: I think one won't work with some default Beat index templates. Two seems very weird, so three seems like the right choice to me. We can discuss separately if there is a need.

=== Metricbeat for Kubernetes monitoring

[source,sh,subs="attributes"]
----
kubectl apply -f {beats_url}/metricbeat_hosts.yaml
----

> 

Reviewer 4: nit: new line

To deploy Beats in clusters with the Pod Security Policy admission controller enabled, or in OpenShift clusters, you must grant additional permissions to the Service Account used by the Beat Pods. Those Service Accounts must be bound to a Role or ClusterRole that has `use` permission for the required Pod Security Policy or Security Context Constraints. Different Beats and their features might require different settings set in their PSP/SCC.

[id="{p}-beat-configuration-examples"]
== Configuration Examples

Below you can find manifests that address a number of common use cases and can be your starting point in exploring Beats deployed with ECK. These manifests are self-contained and work out-of-the-box on any non-secured Kubernetes cluster. All of them contain three-node Elasticsearch cluster and single Kibana instance. All Beat configurations set up Kibana dashboards if they are available for a given Beat and all required RBAC resources.

> [source,sh,subs="attributes"]

Reviewer: ```suggestion

```

=== Metricbeat for Kubernetes monitoring

[source,sh,subs="attributes"]
----
kubectl apply -f {beats_url}/metricbeat_hosts.yaml
----

> Deploys Metricbeat as a DaemonSet that monitors the host resource usage (cpu, memory, network, filesystem) and Kubernetes resources (Nodes, Pods, Containers, Volumes).

Reviewer: ```suggestion
Deploys Metricbeat as a DaemonSet that monitors the host resource usage (CPU, memory, network, filesystem) and Kubernetes resources (Nodes, Pods, Containers, Volumes).
```

=== Filebeat with autodiscover

[source,sh,subs="attributes"]
----
kubectl apply -f {beats_url}/filebeat_autodiscover.yaml
----

> Deploys Filebeat as DaemonSet with autodiscover feature enabled. All pods in all namespaces will have logs shipped to an Elasticsearch cluster.

Reviewer: ```suggestion
Deploys Filebeat as a DaemonSet with the autodiscover feature enabled. It will collect logs from pods in every namespace and load them to the connected Elasticsearch cluster.
```

=== Filebeat with autodiscover for metadata

[source,sh,subs="attributes"]
----
kubectl apply -f {beats_url}/filebeat_autodiscover_by_metadata.yaml
----

> Deploys Filebeat as a DaemonSet with autodiscover feature enabled. Fullfilling any of the two conditions below will cause a given Pod's logs to be shipped to an Elasticsearch cluster:

Reviewer: ```suggestion
Deploys Filebeat as a DaemonSet with the autodiscover feature enabled. Logs from pods matching the following criteria will be shipped to the connected Elasticsearch cluster:
```

- Pod is in `log-namespace` namespace
- Pod has `log-label: "true"` label

=== Filebeat without autodiscover

[source,sh,subs="attributes"]
----
kubectl apply -f {beats_url}/filebeat_no_autodiscover.yaml
----

> Deploys Filebeat as a DaemonSet with autodiscover feature disabled. Uses entire logs directory on the host as the input. Doesn't require any RBAC resources as no Kubernetes APIs are used.

Reviewer: ```suggestion
Deploys Filebeat as a DaemonSet with the autodiscover feature disabled. Uses the entire logs directory on the host as the input source. This configuration does not require any RBAC resources as no Kubernetes APIs are used.
```
